### DISCUSSION TITLE
并击方案受到影响：由Techince维护的小狼毫64位近期修改了 KeyUp时就触发弹出候选框

### DISCUSSION THREAD

**Discussion by rightester** [rime/home#1460] [Q&A] [answered]
我是在这个页面[Techince/weasel/releases/...](https://github.com/Techince/weasel/releases/tag/weasel-0.15.1.0_x64)看到的发布（其中.006版本是上周发布）

其中
> 6、将KeyUp事件弹出候选框改为KeyDown事件弹出候选框。

我认为是该项变动产生了一点新的问题：

并击方案使用了chord_comoposer，可以在未释放最后一个按键时持续改变指法，并有可能该击空指法，不产生输入码——总之，chord_composer的设计是等到最后一个按键释放时，才确定输入码并流出给下一个处理器，才会有后续的产生候选项等步骤

但此新构建的版本（我使用了其中的0.15.1.006）的这项修改使得并击时，候选栏的功能和观感上受到了些破坏。由于候选栏需要等到chord_composer结束才更新候选项，因此在并击过程中，候选栏显示的是上一次并击结束产生的候选项（而如果该次并击直接上屏，则候选栏一直不会更新/清空）

附录屏动图如下
![一小段录屏动图展示](https://github.com/rime/home/assets/33158925/955ec139-0045-4cf8-87d0-9d66eb20e8a3)

对此个人提出个别愚见：

1. 直接完全还原此项更改，恢复原先模式
2. 积极清除候选栏，每次开始新一轮处理击键时都清空一次小狼毫的候选项（非Rime引擎的）
3. 专门对并击合成器修复候选栏，只在用到chord_composer 或正处于并击合成阶段时才等待并击合成完成再更新候选栏地；未使用chord_composer的串击方案可以直接展示/更新；并可以修改chord_composer的实现为有一个击键释放就结束并击合成，以达到减少整体拖沓响应感受的目的

希望后续版本中能修复此问题

（提出此问题是因为并击用户少，担心短期内没有人发现这个问题而被带进未来release，而影响更久远未来的使用
由于不知道应该在哪儿联系项目维护者Techince大佬，看到项目底下说可以在weasel主项目提issue，又看到提issue前应先至讨论区确认，所以在此发帖求助和寻求意见）
（目前小狼毫64位 0.15.1.0版本此功能正常，稍后我就降回该版本了）

**Comment by Sammasc** [comment]
Hi,
Windows 10 Kiosk模式无法使用！有新版本需要解决吗？

**Comment by rightester** [comment, marked as answer]
观察到更新，经实测确认.006版本已修复
> 6、将KeyUp事件弹出候选框改为KeyDown事件弹出候选框。
> 7、最终发现新版小狼毫的DoPaint不会自动响应KeyDown事件，已修复。
